vignettes/fork-and-merge.Rmd

```diff
@@ -0,0 +1,157 @@
+```{r state change2, include=FALSE}
+change_state()
+```
+```{r merging}
+ds <- mergeFork(ds, forked_ds)
+```
+
+After merging the original dataset includes the categorical array variable which we created on the fork. 
+
+```{r check successful merge}
+ds$ImportantHiringCA
+```
+
+It's possible to to make changes to a fork which can't be easily merged into the original dataset. For instance if, while we were working on this fork someone added another variable called `ImportantHiringCA` to the original dataset the merge might fail because there's no safe to reconcile the two forks. This is called a "merge conflict" and there are a couple best practices that you can follow to avoid this problem: 
+
+1) **Make minimal changes to dataset forks.** Instead of making lots of changes to a fork, make a couple of small change to the fork, merge it back into the original dataset, and a create a new fork for the next set of changes
+2) **Have other analysts work on their own forks.** It's easier to avoid conflicts if each member of the team makes changes to their own fork and then periodically merges those changes back into the original dataset. This lets you coordinate the order that you want to apply changes to the original dataset, and so avoid some merge conflicts. 
+
+## Appending data
+
+Another good use of the fork-edit-merge workflow is when you want to append data to an existing dataset. When appending data you usually want to check that the append operation completed successfully before publishing the data to users. This might come up if you are adding a second wave of a survey, or including some additional data which came in after the dataset was initially sent to clients. The first step is to upload the second survey wave as its own dataset. 
+
+```{r state change3, include=FALSE}
+change_state()
+```
+```{r upload wave 2, message=FALSE, results='hide'}
+wave2 <- newDataset(SO_survey, "SO_survey_wave2")
+
+```
+
+We then fork the original dataset and append the new wave onto the forked dataset. 
+```{r state change4, include=FALSE}
+change_state()
+```
+```{r fork-and-append, results='hide'}
+ds_fork <- forkDataset(ds)
+ds_fork <- appendDataset(ds_fork, wave2)
+```
+`ds_fork` now has twice as many rows as `ds` which we can verify with `nrow`:
+
+```{r}
+nrow(ds)
+nrow(ds_fork)
+```
+
+Once we've confirmed that the append completed successfully we can merge the forked dataset back into the original one. 
+```{r state change5, include=FALSE}
+change_state()
+```
+```{r}
+ds <- mergeFork(ds, ds_fork)
+```
+
+`ds` now has the additional rows. 
+
+```{r}
+nrow(ds)
+```
+
+## Merging datasets
+
+Merging two datasets together can often be the source of bugs or data corruption so it's a good candidate for this workflow. Let's create a fake dataset with household size to merge onto the original one. 
```

Comment:
I think we should be a bit more descriptive and change "bugs or data corruption" to "overwriting or misaligning data"